lib/puppet/provider/package/yum.rb

```diff
@@ -159,6 +172,11 @@ def install
         self.debug "Downgrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
         operation = :downgrade
       end
+
+      if is && rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure])) > 0
```

Comment:
Cool, fixed to that. I also changed `is` because that variable name seems too vague